src/Models/VerbSnapshot.php

```diff
@@ -40,4 +40,22 @@ public function scopeWhereDataContains($query, array $data)
     {
         return $query->whereJsonContains('data', $data);
     }
+
+    //
+    // To support custom IDs, we need to override the default Eloquent behavior.
+    //
+
+    public function getKeyType()
+    {
+        return match (config('verbs.id_type', 'snowflake')) {
+            'snowflake' => 'int',
+            'ulid' => 'string',
+            'uuid' => 'string',
+        };
+    }
```

Comment:
I think this should happen on the other two models, too?

Reply:
But why though, sir?

Reply:
lol I guess you're right — sorry! I see that y'all are still working on this and I was just going to resolve my comments and merge it. Probably introduced a merge conflict :(